### DISCUSSION TITLE
InkWell equivalent in Fluent UI

### DISCUSSION THREAD

**Discussion by omensight** [bdlukaa/fluent_ui#928] [Q&A]
Hi, I would like to ask: What is the equivalent of InkWell in Fluent UI for creating a surface with a ripple effect? Thank you.

**Comment by bdlukaa** [comment]
In Windows, there is no such thing as ripple effect. I would direct you to using material's ripple effect or create your own

**Reply by Nico04** [reply]
I'm new to this awesome package, and used a lot InkWell as a base for component that needs to properly react to use input, like a generic button. 
What would be the equivalent in Fluent ?
For instance, What would be a base widget that react like ListTile but with just a onPressed and a child parameter ?
I see that ListTile uses HoverButton internally, but it seems like it needs a lot of states config.

EDIT: [This page](https://bdlukaa.github.io/fluent_ui/#/theming/icons) is exacly what I'm looking for, but looking at [the code](https://github.com/bdlukaa/fluent_ui/blob/85ad5859c136de2f70c7333b7d9308093104bfb5/example/lib/screens/theming/icons.dart#L96) it uses HoverButton and manages states manually. Is there anything simpler ?

EDIT2 : using the basic IconButton is ok, even if naming is quite wrong for the use case. Why didn't you use the IconButton widget on the Theming/Icons page of your example ?

**Reply by bdlukaa** [reply]
> Why didn't you use the IconButton widget on the Theming/Icons page of your example ?

`IconButton` has constraints. The best option was a custom widget.
